### ISSUE TITLE
-S and -cc combination doesn't work

### ISSUE THREAD

**Issue by edshamis** [mountaineerbr/shellChatGPT#15]
Looks like `-S <file> -cc` don't work together, the chat is started but instruction file is not taken into consideration.

**Comment by mountaineerbr** [comment]
Hello again, my friend!

I could not catch that error. I have been updating this script a lot in the last few days.. Did you test it with the latest version already?

I could not really find that error. I have changed the code a little bit in the last few days, and it is not printing INSTRUCTIONS bu defaults any more (as `-cc` is  chat options, it was defaulting to _much_ _less_ verbose).

As per the follwoing test with `v0.12.7`, it is loading the text file:

```
% print "Be a nice bot!" > test.txt130 jsn@archhost ~ % print "Be a nice bot\!" > test.txt
me % chatgpt.sh -VVcc -S test.txt Hello\!
SESSION BREAK
chat/completions
{"messages":[{"role":"system","content":"Be a nice bot!"},{"role":"user","content":"Hello!"}],"model":"gpt-3.5-turbo","temperature":0.9,"presence_penalty":0.6,"max_tokens":256,"n":1}
<CTRL-D> redo, <CTR-C> exit, or continue
```

OK, so based on this conversation, I set INSTRUCTION to be printed always at the start of a new session (independently of verbose `options -vvv`) because it is very important.

May I ask you what distro you run?

Thanks for all the feedback, really!

**Comment by mountaineerbr** [comment]
Hey there, Ed! I love it when you bring me 🐞🐞!!

The script was not setting max tokens for gpt-4 and gpt-4-32k models because I had mis-spelled them (as gpt4, without the hyphen) in the case statement where it should be automatically set, [see it here](https://github.com/mountaineerbr/shellChatGPT/blob/6a5d8a2a9cee6bacc923a775a8aa1d8e96b77958/chatgpt.sh#L1967-L1968).


Thanks for catching this bug! I will upload a fix in `v0.12.10` after I post this answer. So you can use it right away.

---

Below are some thoughts on reviewing token count code. This is just for the record. No need to read it.

One can set model capacity manually to 8192tkns (and request max to 512tkns) with `chatgpt.sh -8192,512`.. In this case, user must supply two values otherwise he only sets max response tokens (or the function may just set it to model capacity and set previous model capacity to response max, which will stop the model itself)..

It turns out that token count may be a problem with continuing big inputs (if all input prompts are very big and only a few of them can be included in the history context), as we are currently using a shell hack to count tokens in input in a _very rough manner_... I already implemented the proper python tiktoken lib in a function to do proper token count, but it is slower to count tokens in small input, and faster with bigger input... So, currently, I am at a loss about what to do with regards to counting tokens.
